src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -10,29 +10,103 @@ trait SortedSets {
 
   final val zadd = Command(
     "ZADD",
-    Tuple6(
+    Tuple5(
       StringInput,
       OptionalInput(ZAddUpdateInput),
       OptionalInput(ZAddChangedInput),
       OptionalInput(ZAddIncrementInput),
-      LongInput,
-      NonEmptyList(StringInput)
+      NonEmptyList(ZAddMemberScoresInput)
     ),
     LongOutput
   )
 
   final val zcard   = Command("ZCARD", StringInput, LongOutput)
   final val zcount  = Command("ZCOUNT", Tuple2(StringInput, RangeInput), LongOutput)
   final val zincrby = Command("ZINCRBY", Tuple3(StringInput, LongInput, ByteInput), LongOutput)
+
+  final val zlexcount = Command("ZLEXCOUNT", Tuple2(StringInput, ZLexCountInput), LongOutput)
+
+  final val zinterstore = Command(
+    "ZINTERSTORE",
+    Tuple3(
```

Comment:
You can make it a `Tuple4` and separate keys from weights. Later on, one syntax class needs to be added that has curried parameters to support parameters after non-empty list.

Reply:
The idea was to represent the fact that `ZINTERSTORE` and `ZUNIONSTORE` can only either have a non-empty list of keys or a non-empty list of key/weight pairs using the type system. I did away with the `numkeys` parameter, since I thought it would be given by the list itself. However as far as I understood... I misunderstood and you have something else in mind. :-)
I have splitted keys and weights in the next version.

Reply:
Ah, but they can both be present, see the example from command docs: `ZUNIONSTORE out 2 zset1 zset2 WEIGHTS 2 3` :).

Reply:
You're right about `numkeys` though. That one's redis being "silly" :).

Reply:
As discussed we will settle on this version for `ZUNIONSTORE` (and `ZINTERSTORE`) for now:
```
  final val zunionstore = Command(
    "ZUNIONSTORE",
    Tuple4(
      StringInput,
      NonEmptyList(StringInput),
      OptionalInput(NonEmptyList(DoubleInput)),
      OptionalInput(SortedSetAggregateInput)
    ),
    LongOutput
  )
```